source/s3_client.c

```diff
@@ -1967,6 +1987,15 @@ static void s_s3_client_body_streaming_task(struct aws_task *task, void *arg, en
     aws_s3_client_release(client);
 }
 
+/* Called by aws_s3_request when it has finished being destroyed */
+static void s_s3_client_clear_waiting_for_first_host_resolve_callback(struct aws_s3_client *client) {
```

Comment:
***_synced? Keep the naming consistent with other function that touches the synced data.